### PR TITLE
Counter measure sophos and palo

### PR DESCRIPTION
**Pull request by mahirchavda**

### CODE REVIEW

File: cyences_app_for_splunk/appserver/static/cs_cm_palo_ip_block.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        CMUtils.disableButton(BLOCK_IP_BUTTON_ID);
+        CMUtils.disableButton(ALLOW_IP_BUTTON_ID);
+    }
+
+    function enableButtons() {
+        CMUtils.enableButton(BLOCK_IP_BUTTON_ID);
+        CMUtils.enableButton(ALLOW_IP_BUTTON_ID);
+    }
+
+    async function paloBlockIPEndpoint(){
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/appserver/static/cs_cm_palo_ip_block.js:36]: paloBlockIP()

File: cyences_app_for_splunk/appserver/static/cs_cm_palo_ip_block.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        try{
+            res = await executeAsyncRestCall(`/CounterMeasurePaloIPBlock/${paloAction}`, 'POST', null, data);
+            return [true, `The ${ipAddress} has been successfully ${paloAction}ed.`];
+        }
+        catch(err){
+            return [false, `Unexpected error occurred. Error: ${err}`]
+        }
+    }
+
+    async function paloBlockIPEndpointCallbackFunction(username, password){
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/appserver/static/cs_cm_palo_ip_block.js:55]: paloBlockIPCallbackFunction

File: cyences_app_for_splunk/appserver/static/cs_cm_palo_ip_block.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        paloUsername = username;
+        paloPassword = password;
+        return await paloBlockIPEndpoint();
+    }
+
+
+    async function paloBlockIPEndpointButtonClickEventHandler(){
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/appserver/static/cs_cm_palo_ip_block.js:62]: paloBlockIPButtonClickEventHandler

File: cyences_app_for_splunk/appserver/static/cs_cm_utils.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+        /* Common Credential Modal Functionality */
+        static addModalToHTML(id, usernamePlaceholder="Username", passwordPlaceholder="Password"){
+            let credentialModalHTMLTemplate = `<p id="${id}_message"/>
+            <span class="${id}_loader"/>
+            <div class="modal fade" id="${id}_myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true" style="display:none;">
+            <div class="modal-dialog" role="document">
+            <div class="modal-content">
+                <div class="modal-header text-center">
+                <h4 class="modal-title w-100 font-weight-bold">Sophos Endpoint Protection Authorization</h4>
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/appserver/static/cs_cm_utils.js:25]: Need to make the label dynamic. "Sophos Endpoint Protection Authorization"

File: cyences_app_for_splunk/default/data/ui/nav/default.xml [hunk 5]
```diff
@@ -31,6 +31,10 @@
     <view name="cs_lansweeper" />
     <view name="cs_vulnerability" />
   </collection>
+  <collection label="Counter Measures">
+    <view name="cs_cm_sophos_endpoints" />
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/data/ui/nav/default.xml:35]: We'll remove this from the navigation before merging the code.

File: cyences_app_for_splunk/default/web.conf [hunk 6]
```diff
@@ -20,4 +24,8 @@ methods = GET,POST
 
 [expose:SophosEndpointConfiguration_specified]
 pattern = SophosEndpointConfiguration/*
-methods = GET,POST
\ No newline at end of file
+methods = GET,POST
+
+[expose:CounterMeasureSophosEndpoint_specified]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/web.conf:29]: Need to add endpoint for palo firewall.

File: cyences_app_for_splunk/default/restmap.conf [hunk 7]
```diff
@@ -18,4 +18,10 @@ python.version = python3
 handlertype = python
 handlerfile = rh_sophos_endpoint_conf.py
 handleractions = list,edit
-python.version = python3
\ No newline at end of file
+python.version = python3
+
+[admin_external:CounterMeasureSophosEndpoint]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/restmap.conf:23]: Need to add endpoint for palo firewall.

File: cyences_app_for_splunk/default/data/ui/views/cs_cm_palo_alto_firewall_ip_block.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<form theme="dark" version="1.1" script="cs_cm_palo_ip_block.js" >
+  <label>Palo Alto Firewall IP Block - Counter Measures</label>
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/data/ui/views/cs_cm_palo_alto_firewall_ip_block.xml:2]: Counter Measures - Palo Alto Firewall - IP Block/Allow

File: cyences_app_for_splunk/bin/sophos_endpoint_api_utils.py [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+
+                current_page = current_page + 1
+
+
+    def isolate_endpoint(self, uuid, comment):
+        if uuid=="" or uuid==" ":
+            raise Exception("No endpoint_uuid provided.")
+
+        for tenant in self.sophos_tenant_dict:
+            self.logger.debug("checking under tenant:{}".format(tenant))
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/bin/sophos_endpoint_api_utils.py:209]: Need to check and remove unnecessory debug messages before merging.

**Comment by mahirchavda** [comment]
Not required

**Pull request closed without merging**